lib/create_response.js

```diff
@@ -65,7 +66,11 @@ function createResponse(message) {
                 }
               },
               error => {
-                error ? controller.error(error) : controller.close()
+                if (error) {
+                  controller.error(error)
+                } else if (!isCanceled) {
+                  controller.close()
+                }
```

Comment:
I was able to reproduce the error on Mac OS in Node 21. The else path here was not taken

<img width="506" alt="image" src="https://github.com/nock/nock/assets/39992/c643f261-7894-46ef-8d1d-ee0712cea985">

Could it somehow be a race condition?

Reply:
If I see correctly, the branch was taken 7 times.. 
Maybe it never gets `isCanceled=true`?

Reply:
> the branch was taken 7 times

you are right 🤔 maybe something is wrong with the coverage tooling

Reply:
I think it's because there is no test that `isCancled` is true, so it never goes past the `if else` (to oblivion lol).
I'm wondering why it's flaky, and I guess it has something to do with the v8 and its different compilers (but again, it's just a wild guess..)

Reply:
yes, your are right